### ISSUE TITLE
Wrong error message: Super weird behavior of f_front

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#547]
This small example stripped down from a >3000 LOC file gives a weird error message and leads user into wrong assumptions. 

```fortran
MODULE mod1
  IMPLICIT NONE
CONTAINS
  SUBROUTINE sub1()
    IMPLICIT NONE

    REAL :: varLevel1

  CONTAINS
    SUBROUTINE subsub1()
      REAL :: varLevel2

      varLevel1 = 1.5  ! 1st assignment 
      varLevel2 = varLevel1 * fct1(1.2) ! 2nd assignment
    END SUBROUTINE subsub1
  END SUBROUTINE sub1
END MODULE mod1
```

```bash
"example.f90", line 16: attempt to use undefined type variable, varlevel1
```

The variable `varLevel1` is correctly declared but the `fct1` is not. `F_Front` should complain about this normally. 

What is really weird is that is the line `2nd assignment` line is commented, this code is successfully parsed by `F_Front`. On the other hand, if we comment the `1st assignment` line, the code is also parsed successfully. 

This is very confusing. 

`gfortran -c` would give a comprehensive error msg. 
```bash
varLevel2 = varLevel1 * fct1(1.2)
                                  1
Error: Function 'fct1' at (1) has no IMPLICIT type
```
